### PR TITLE
API reference DDS DCPS PIM [8336]

### PR DESCRIPTION
**Pull request by EduPonz**
DDS DCPS PIM API reference

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/244/)

File: docs/fastdds/api_reference/spelling_wordlist.txt [hunk 1]
```diff
@@ -85,3 +85,34 @@ statusmask
 PublicationMatchedStatus
 statusmask
 SubscriptionMatchedStatus
+Unregister
+Otherwhise
```
Comment (IkerLuengo) [inline docs/fastdds/api_reference/spelling_wordlist.txt]: Not a valid word

File: docs/fastdds/api_reference/spelling_wordlist.txt [hunk 2]
```diff
@@ -85,3 +85,34 @@ statusmask
...
+nullptr
+differnet
```
Comment (IkerLuengo) [inline docs/fastdds/api_reference/spelling_wordlist.txt]: Not a valid word

File: docs/fastdds/api_reference/spelling_wordlist.txt [hunk 3]
```diff
@@ -85,3 +85,34 @@ statusmask
...
+getTypeDependencies
+getTypes
+typelookup
+enought
```
Comment (IkerLuengo) [inline docs/fastdds/api_reference/spelling_wordlist.txt]: Not a valid word

File: docs/fastdds/api_reference/spelling_wordlist.txt [hunk 4]
```diff
@@ -85,3 +85,34 @@ statusmask
...
+typelookup
```
Comment (IkerLuengo) [inline docs/fastdds/api_reference/spelling_wordlist.txt]: Repeated

File: docs/fastdds/api_reference/spelling_wordlist.txt [hunk 5]
```diff
@@ -85,3 +85,34 @@ statusmask
...
+DomainParticipantListener
+NoOpDomainParticipantListener
+dyn
+ptr
+fastrtps
+TypeIdentifier
+params
+livelivess
```
Comment (IkerLuengo) [inline docs/fastdds/api_reference/spelling_wordlist.txt]: Not a valid word.

File: docs/fastdds/api_reference/spelling_wordlist.txt [hunk 6]
```diff
@@ -85,3 +85,34 @@ statusmask
...
+datawriter
+datawriter
```
Comment (IkerLuengo) [inline docs/fastdds/api_reference/spelling_wordlist.txt]: Repeated (and some more down below)

File: docs/fastdds/api_reference/spelling_wordlist.txt [hunk 7]
```diff
@@ -85,3 +85,34 @@ statusmask
...
+unregister
+datawriters
+timedout
```
Comment (IkerLuengo) [inline docs/fastdds/api_reference/spelling_wordlist.txt]: Probably should be cahnged to `timed ou` whenever it appears on the docs

File: docs/fastdds/api_reference/spelling_wordlist.txt (hunk 1 above)
Comment (EduPonz) [reply docs/fastdds/api_reference/spelling_wordlist.txt]: As a reply to all the comments:
1. I think you're not looking at the last commit, I believe I've taken care of all duplicates
2. Actually a lot of wrong words made the list. The problem is that they are wrong in the source code doxygen documentation of the header files themselves. I'd like to have them corrected in Fast itself, but until then I've added them to the dictionary so we don't ge hundreds of warnings we are not causing.

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/245/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/246/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/247/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/254/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/255/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/256/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/265/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/271/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/273/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/274/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/275/)

File: docs/fastdds/api_reference/includes/dds_pim/core/status/livelinessloststatus.rst [hunk 8]
```diff
@@ -0,0 +1,7 @@
+.. _api_pim_rtpslivelinessloststatus:
+
+RTPSLivelinessLostStatus
+------------------------
+
+.. doxygentypedef:: eprosima::fastrtps::LivelinessLostStatus
+    :project: Fast RTPS
```
Comment (laura-eprosima) [inline docs/fastdds/api_reference/includes/dds_pim/core/status/livelinessloststatus.rst:7]: This is a fastrtps status, so it is not needed to include it

Comment (EduPonz) [reply docs/fastdds/api_reference/includes/dds_pim/core/status/livelinessloststatus.rst]: done

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/285/)

**Pull request merged**